src/api/spec/components/previews/bs_request_action_description_component_preview.rb

```diff
@@ -7,6 +7,6 @@ def submit_preview
 
   def add_role_preview
     bs_request = BsRequest.joins(:bs_request_actions).where(bs_request_actions: { type: :add_role }).last
-    render(BsRequestActionDescriptionComponent.new(action: bs_request.webui_actions.find { |a| a[:type] == :add_role }, creator: bs_request.creator))
+    render(BsRequestActionDescriptionComponent.new(action: bs_request.bs_request_actions.first { |a| a[:type] == 'add_role' }))
```

Comment:
It's not working as it is simply returning the first action (not the first with `add_role`) . We can replace it with:

```suggestion
    render(BsRequestActionDescriptionComponent.new(action: bs_request.bs_request_actions.where(type: 'add_role').first))
```

Reply:
You are right, adding...